pkg/agent/nodeportlocal/portcache/port_table_linux.go

```diff
@@ -271,20 +190,24 @@ func (pt *PortTable) DeleteRulesForPod(podIP string) error {
 func (pt *PortTable) syncRules() error {
 	pt.tableLock.Lock()
 	defer pt.tableLock.Unlock()
-	nplPorts := make([]rules.PodNodePort, 0, len(pt.NodePortTable))
-	for _, npData := range pt.NodePortTable {
-		protocols := make([]string, 0, len(supportedProtocols))
-		for _, protocol := range npData.Protocols {
+	nplPorts := make([]rules.PodNodePort, 0, 1)
```

Comment:
List data from the cache directly?

Reply:
> List data from the cache directly?

No, because we can't change to data type of the cache slices directly.

Reply:
> Why do you set the slice max as 1 ?

Updated.